README.md
# extractISO

Just a little script to extract a ISO in a directory.

Need the p7zip package

Need the zenity package (for zenity version).


## Test from the git source
```
$ git clone https://github.com/CyrilleBiot/extractISO.git
$ cd extractISO/
```

## Shell version
```
$ ./source/extract-iso.sh
```

## Zenity version
```
$ ./source/extract-iso-zenity.sh
```

## Install the deb package
```
# dpkg -i extract-iso_<CHOOSE-YOUR-VERSION>_all.deb
```



debian/changelog
extract-iso (0.1) unstable; urgency=medium

  * Initial release.

 --  <ragnarok@libres>  Sat, 26 Sep 2020 22:54:29 +0200


extractISO-zenity.sh
#!/bin/sh
#
#__Source : https://github.com/CyrilleBiot/extractISO
#__author__ = "Cyrille BIOT <cyrille@cbiot.fr>"
#__copyright__ = "Copyleft"
#__site__ = "https://cbiot.fr"
#__license__ = "GPL"
#__version__ = "1.0"
#__email__ = "cyrille@cbiot.fr"
#__status__ = "Devel"
#__date__ = "2020/09/26"
#
#

# ====================================================
# Test p7zip / installé ?
if which p7zip > /dev/null; then
    echo "Cool, p7zip est installé."
else
    echo "Il vous faut installer p7zip"
	echo "Soit : # apt install p7zip"
	echo "Soit : $ sudo # apt install p7zip"
	exit 0
fi

# Test zenity / installé ?
if which zenity > /dev/null; then
    echo "Cool, zenity est installé."
else
    echo "Il vous faut installer zenity"
	echo "Soit : # apt install zenity"
	echo "Soit : $ sudo # apt install zenity"
	exit 0
fi
# ====================================================


# Lancement de zenity / Fenetre d'accueil
  zenity  --info --title="Extraction d'ISO" --text="Juste un extracteur d ISO vers un répertoire.


            Etape 1 : choisir son ISO.


            Etape 2 : Séléctionner répertoire de destination


	    Source : https://github.com/CyrilleBiot/extractISO 

            Mail : cyrille@cbiot.fr" --width=600

		 
# Recuperation de l'iso
  FILE=$(zenity --file-selection --file-filter='Image ISO (.iso) | *.iso' --title="Selectionnez votre image ISO")
  echo "Fichier ISO sélectionné : $FILE"

# Localication du dossier de destination
  DEST=$(zenity --file-selection --directory --title="Sélectionner le répertoire de detination")
  echo "Repertoire de destination : $DEST"

# Test du répetoire d'extraction
  if [ -d "$DEST" ]; then
	echo "Le dossier existe, on le vide."
	rm -rf $DEST
	mkdir $DEST
  else 
	echo "Le dossier n'existe pas, on le crée."
	mkdir $DEST
  fi

# Extraction de l'ISO
  # On entre dans le répertoire de destination
  cd $DEST
  # On extrait l'iso
  7z x -y $FILE
  # On liste son contenu
  ls



extractISO/extractISO.sh
#!/bin/sh
#
#__Source : https://github.com/CyrilleBiot/extractISO
#__author__ = "Cyrille BIOT <cyrille@cbiot.fr>"
#__copyright__ = "Copyleft"
#__site__ = "https://cbiot.fr"
#__license__ = "GPL"
#__version__ = "1.0"
#__email__ = "cyrille@cbiot.fr"
#__status__ = "Devel"
#__date__ = "2020/09/26"
#
#

FILE=$1
DEST=$2

# Fonction pour les messages à afficher
message () {
	case "$1" in
	  # Usage	
      -help)
         echo "Usage: sh extractISO.sh /chemin/jusque/monFichierFormatISO.iso /dossier/ou/extraire/cette/iso" 
         ;;

      # ISO présente
      -isofound)
		 echo "Une image ISO a bien été trouvée."
		 ;;
     # ISO présente
	  -isonotfound)
		  echo "Aucune image ISO trouvée."
		  echo "Vérifier la présence de celle-ci et la syntaxe de votre paramètre 1"
		  ;;
	  # Rep DEST trouvé, supprimer
	  -dirfound)
	     echo "Le répertoire de destination existe. On le supprime."
	     ;;
	  # Rep DEST,création  
	  -drinotfound)
	     echo "Le répertoire de destination n'existe pas. On le crée."
		 ;;
      *)  # No more options 
      ;;
    esac
}


# Afficher aide
if [ "$1" = "-h" -o "$1" = "--help" ] ; then
    message -1
    exit 0
fi


# Test p7zip / installé ?
if which p7zip > /dev/null; then
    echo "Cool, p7zip est installé."
else
    echo "Il vous faut installer p7zip"
	echo "Soit : # apt install p7zip"
	echo "Soit : $ sudo # apt install p7zip"
	exit 0
fi


# Test de la validité nbre paramètres
if [ $# -ne 2 ]; then
	echo "Nombre de paramètres incorrects."
    message -help
    exit 0
fi


# Test de la présence du fichier ISO
if [ -f "$FILE" ]; then
    message -isofound
else
    message -isonotfound
    exit
fi


# Test du répetoire d'extraction
if [ -d "$DEST" ]; then
	message -dirfound
	rm -rf $DEST
	mkdir $DEST
else 
	message -dirnotfound
	mkdir $DEST
fi

# Extraction de l'ISO
  # On entre dans le répertoire de destination
  cd $2
  # On extrait l'iso
  7z x -y $1
  # On liste son contenu
  ls


source/extract-iso.sh
#!/bin/sh
#
#__Source : https://github.com/CyrilleBiot/extractISO
#__author__ = "Cyrille BIOT <cyrille@cbiot.fr>"
#__copyright__ = "Copyleft"
#__site__ = "https://cbiot.fr"
#__license__ = "GPL"
#__version__ = "1.0"
#__email__ = "cyrille@cbiot.fr"
#__status__ = "Devel"
#__date__ = "2020/09/26"
#
#

FILE=$1
DEST=$2

# Fonction pour les messages à afficher
message () {
	case "$1" in
	  # Usage	
      -help)
         echo "Usage: sh extractISO.sh /chemin/jusque/monFichierFormatISO.iso /dossier/ou/extraire/cette/iso" 
         ;;

      # ISO présente
      -isofound)
		 echo "Une image ISO a bien été trouvée."
		 ;;
     # ISO présente
	  -isonotfound)
		  echo "Aucune image ISO trouvée."
		  echo "Vérifier la présence de celle-ci et la syntaxe de votre paramètre 1"
		  ;;
	  # Rep DEST trouvé, supprimer
	  -dirfound)
	     echo "Le répertoire de destination existe. On le supprime."
	     ;;
	  # Rep DEST,création  
	  -drinotfound)
	     echo "Le répertoire de destination n'existe pas. On le crée."
		 ;;
      *)  # No more options 
      ;;
    esac
}


# Afficher aide
if [ "$1" = "-h" -o "$1" = "--help" ] ; then
    message -1
    exit 0
fi


# Test p7zip / installé ?
if which p7zip > /dev/null; then
    echo "Cool, p7zip est installé."
else
    echo "Il vous faut installer p7zip"
	echo "Soit : # apt install p7zip"
	echo "Soit : $ sudo # apt install p7zip"
	exit 0
fi


# Test de la validité nbre paramètres
if [ $# -ne 2 ]; then
	echo "Nombre de paramètres incorrects."
    message -help
    exit 0
fi


# Test de la présence du fichier ISO
if [ -f "$FILE" ]; then
    message -isofound
else
    message -isonotfound
    exit
fi


# Test du répetoire d'extraction
if [ -d "$DEST" ]; then
	message -dirfound
	rm -rf $DEST
	mkdir $DEST
else 
	message -dirnotfound
	mkdir $DEST
fi

# Extraction de l'ISO
  # On entre dans le répertoire de destination
  cd $DEST
  # On extrait l'iso
  7z x -y $FILE
  # On liste son contenu
  ls
